src/code_809A560.c

```diff
@@ -184,41 +285,1788 @@ void sub_809A7EC(void)
+                    gUnknown_3001B64->unk430 = -1;
+                    return 0;
+                }
+            }
+            else {
+                switch(sub_801A6E8(1)) {
+                    case 3: {
+                        s32 itemArrayId = sub_801A8AC();
+                        s32 itemId = gTeamInventoryRef->teamItems[itemArrayId].id;
+                        if (IsEdibleItem(itemId)) {
+                            BufferItemName(gFormatBuffer_Items[0],itemId,0);
+                            ShiftItemsDownFrom(itemArrayId);
+                            gUnknown_3001B64->unk430 = 1;
+                        }
+                        else {
+                            gUnknown_3001B64->unk430 = -1;
+                        }
+                        sub_801A928();
+                        return 0;
+                    }
+                    case 2:
+                        gUnknown_3001B64->unk430 = -1;sub_801A928();return 0;
+                        return 0;
+                    default:
+                        return 1;
+                }
+            }
+            break;
+        case 0x2E:
+            if (gUnknown_3001B64->unk420 == 1) {
+                ResetTextbox_809B294();
+                if (DrawCredits(gUnknown_3001B64->unk424, gUnknown_3001B64->unk428)) {
+                    return 1;
+                }
+                gUnknown_3001B64->unk430 = -1;
+                return 0;
+            }
+            else {
+                switch (sub_8035574()) {
+                    case 2:
+                    case 3:
+                        gUnknown_3001B64->unk430 = -1;
+                        sub_803565C();
+                        return 0;
+                    default:
+                        return 1;
+                }
+            }
+            break;
+    }
+
+    return 0;
+}
+
+void sub_809C39C(void)
+{
+    gUnknown_3001B64->unk430 = sub_801D178();
+    if(gUnknown_3001B64->unk430 == 3)
+        GroundMainGroundRequest(sub_8002694(sub_801D1D4()), 0, -1);
+    sub_801D1E0();
+}
+
+void sub_809C3D8(void)
+{
+    gUnknown_3001B64->unk430 = sub_801D178();
+    if(gUnknown_3001B64->unk430 == 3)
+        GroundMainGroundRequest(sub_8002694(sub_801D1D4()), 0, -1);
+    sub_801D1E0();
+}
+
+void sub_809C414(void)
+{
+    u8 val;
+
+    val = sub_802540C();
+
+    if(val != 0)
+    {
+        gUnknown_3001B64->unk430 = val;
+        GroundMainGroundRequest(sub_8002694(val), 0, -1);
+    }
+    else
+{
+        gUnknown_3001B64->unk430 = -1;
+    }
+    CleanFriendListMenu();
+
+}
+
+void sub_809C464(void)
+{
+	DeleteTestTracker();
+	sub_8001064();
+	sub_809965C();
+}
+
+void sub_809C478(void)
+{
+    s32 temp;
+    s32 temp1;
+
+    if(sub_80356A0() == 0x3)
+    {
+        PlaySound(0x313);
+        InitFlash();
+        ClearFlashData();
+    }
+
+    sub_8035758();
+
+    GetScriptVarScenario(SCENARIO_MAIN, &temp, &temp1);
+    sub_80993C0(0);
+}
+
+// https://decomp.me/scratch/WyHAL  - (84.32% matching - Seth)
```

Comment:
Can probably sub with this scratch by @Kermalis who is closer

https://decomp.me/scratch/pTEda

Reply:
Thanks to Kermalis' scratch I matched the function 😄